src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsProvider.cs

```diff
@@ -164,17 +168,38 @@ protected set
             }
         }
 
+        public override NamedIdentity DataSourceKey { get; } = new NamedIdentity(nameof(LaunchSettingsProvider));
+
+        // _nextVersion represents the version we will use in the future, so we need to
+        // subtract 1 to get the current version.
+        public override IComparable DataSourceVersion => _nextVersion - 1;
+
+        public override IReceivableSourceBlock<IProjectVersionedValue<ILaunchSettings>> SourceBlock
+        {
+            get
+            {
+                EnsureInitialized();
+                return _versionedChangedSourceBlock!;
+            }
+        }
+
         /// <summary>
         /// The LaunchSettingsProvider sinks 2 sets of information:
         /// 1. Changes to the launchsettings.json file on disk
         /// 2. Changes to the ActiveDebugProfile property in the .user file
         /// </summary>
         protected override void Initialize()
         {
-            // Create our broadcast block for subscribers to get new ILaunchProfiles Information
+            base.Initialize();
+
+            // Create our broadcast block for subscribers to ILaunchSettingsProvider to get new ILaunchProfiles information
```

Comment:
The comment mentioned `ILaunchProfiles` but the code mentions `ILaunchSettings`. Is that right?

Reply:
Yeah that should be `ILaunchSettings`.

Reply:
Fixed.